baybe/surrogates/base.py

```diff
@@ -189,71 +208,19 @@ def wrapper(obj):
     return wrapper
 
 
-# >>>>>>>>>>>>>>>>>>>>>>>>>>>>>> Temporary workaround >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
-def _structure_surrogate(val, _):
-    """Structure a surrogate model."""
-    # TODO [15436]
-    # https://***REMOVED***/_boards/board/t/SDK%20Devs/Features/?workitem=15436
-
-    # NOTE:
-    # Due to above issue,
-    # it is difficult to find the wrapped class in the subclass structure.
-    # The renaming only happens in the init method of wrapper classes
-    # (classes that haven't been initialized won't have the overwritten name)
-    # Since any method revolving `cls()` will not work as expected,
-    # we rely temporarily on `getattr` to allow the wrappers to be called on demand.
-
-    _type = val["type"]
-
-    cls = getattr(sys.modules[__package__], _type, None)
-    # cls = getattr(baybe.surrogates, ...) if used in another module
-
-    if cls is None:
-        raise ValueError(f"Unknown subclass {_type}.")
-
-    # NOTE:
-    # This is a workaround for onnx str type being `str` or `bytes`
-    onnx_str = val.get("onnx_str", None)
-    if onnx_str and isinstance(onnx_str, str):
-        val["onnx_str"] = onnx_str.encode(_ONNX_ENCODING)
-
-    return converter.structure_attrs_fromdict(val, cls)
-
-
-def get_available_surrogates() -> list[type[Surrogate]]:
-    """List all available surrogate models.
-
-    Returns:
-        A list of available surrogate classes.
-    """
-    # List available names
-    available_names = {
-        cl.__name__
-        for cl in get_subclasses(Surrogate)
-        if cl.__name__ not in _WRAPPER_MODELS
-    }
-
-    # Convert them to classes
-    available_classes = [
-        getattr(sys.modules[__package__], mdl_name, None)
-        for mdl_name in available_names
-    ]
-
-    # TODO: The initialization of the classes is currently necessary for the renaming
-    #  to take place (see [15436] and NOTE in `structure_surrogate`).
-    [cl() for cl in available_classes if cl is not None]
-
-    return [cl for cl in available_classes if cl is not None]
-
-
 # Register (un-)structure hooks
-# TODO: Needs to be refactored
+# IMPROVE: Ideally, the ONNX-specific hooks should simply be registered with the ONNX
+#   class, which would avoid the nested wrapping below. However, this requires
+#   adjusting the base class (un-)structure hooks such that they consistently apply
+#   existing hooks of the concrete subclasses.
 converter.register_unstructure_hook(
     Surrogate,
-    _decode_onnx_str(_block_serialize_custom_architecture(unstructure_base)),
+    _decode_onnx_str(
+        _block_serialize_custom_architecture(
+            lambda x: unstructure_base(x, overrides={"_model": override(omit=True)})
+        )
+    ),
+)
+converter.register_structure_hook(
```

Comment:
I think the confusion in terminology here lies in the fact that we are talking about different levels. What `encoding` refers to in this context is that the internal string gets encoded prior to passing it to the actual hook, see function body:
<img width="657" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/69465c6a-7c3f-4209-99da-06139016e250">
Does this answer the question?

Reply:
Kind of, but I am still not happy with the naming here, since the `_encode_onnx_str` does not just encode a `str`, but also returns the hook. For me, the name of the function reads that "I pass an ONNX string, and this function encodes this string", but it does something completely different: It returns a hook that internally encodes a string into ONNX format (that is at least how I understand the code)

Reply:
Agree with your criticism. Please be aware thought, that the code existed already before the PR (back then written by Elex) and I merely separated it out of the workaround code. Still, will fix it and ping you here

Reply:
@AVHopp Done. Please unresolve if further comments